ptf/saiwred.py

```diff
@@ -0,0 +1,431 @@
+# Copyright 2021-present Intel Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Thrift SAI interface Weighted random early detection (WRED) tests
+"""
+from sai_thrift.sai_headers import *
+
+from sai_base_test import *
+
+
+@group("draft")
+class WredBaseTest(SaiHelper):
+    """
+    WRED base test
+    """
+    def setUp(self):
+        """
+        Configure WredTest:
+
+        1. Call SaiHelper setUp()
+        2. Configure routing.
+           - route traffic destined for 172.20.20.11 into port 11
+           - route traffic destined for 172.20.20.12 into port 12
+           - route traffic destined for 4000::2 into port 13
+        3. Get queues so that they can be used within tests
+           - self.queues10 of port 10
+           - self.queues11 of port 11
+           - self.queues12 of port 12
+           - self.queues13 of port 13
```

Comment:
We also need to configure DSCP or PCP to TC and TC to queue mapping to configure WRED profiles to non-default queues. Please add them to the setup